File: api-tester/src/main/java/com/revenuecat/apitester/java/SubscriptionOptionAPI.java
package com.revenuecat.apitester.java;

import com.android.billingclient.api.ProductDetails;
import com.revenuecat.purchases.PresentedOfferingContext;
import com.revenuecat.purchases.models.GoogleSubscriptionOption;
import com.revenuecat.purchases.models.PricingPhase;
import com.revenuecat.purchases.models.PurchasingData;
import com.revenuecat.purchases.models.SubscriptionOption;

import java.util.List;

@SuppressWarnings({"unused"})
final class SubscriptionOptionAPI {

    static void checkSubscriptionOption(SubscriptionOption subscriptionOption) {
        List<PricingPhase> pricingPhases = subscriptionOption.getPricingPhases();
        List<String> tags = subscriptionOption.getTags();
        Boolean isBasePlan = subscriptionOption.isBasePlan();
        String presentedOfferingId = subscriptionOption.getPresentedOfferingIdentifier();
        PresentedOfferingContext presentedOfferingContext = subscriptionOption.getPresentedOfferingContext();
        PurchasingData purchasingData = subscriptionOption.getPurchasingData();
        String id = subscriptionOption.getId();
        Boolean isPrepaid = subscriptionOption.isPrepaid();
    }

    static void checkGoogleSubscriptionOption(GoogleSubscriptionOption googleSubscriptionOption) {
        checkSubscriptionOption(googleSubscriptionOption);
        String productId = googleSubscriptionOption.getProductId();
        String basePlanId = googleSubscriptionOption.getBasePlanId();
        String offerId = googleSubscriptionOption.getOfferId();
        String offerToken = googleSubscriptionOption.getOfferToken();
        ProductDetails productDetails = googleSubscriptionOption.getProductDetails();

        GoogleSubscriptionOption constructedGoogleSubOption = new GoogleSubscriptionOption(
                productId,
                basePlanId,
                offerId,
                googleSubscriptionOption.getPricingPhases(),
                googleSubscriptionOption.getTags(),
                productDetails,
                offerToken
        );

        GoogleSubscriptionOption constructedGoogleSubOptionWithOfferingId = new GoogleSubscriptionOption(
                productId,
                basePlanId,
                offerId,
                googleSubscriptionOption.getPricingPhases(),
                googleSubscriptionOption.getTags(),
                productDetails,
                offerToken,
                googleSubscriptionOption.getPresentedOfferingIdentifier()
        );

        GoogleSubscriptionOption constructedGoogleSubOptionWithPresentedContext = new GoogleSubscriptionOption(
                productId,
                basePlanId,
                offerId,
                googleSubscriptionOption.getPricingPhases(),
                googleSubscriptionOption.getTags(),
                productDetails,
                offerToken,
                googleSubscriptionOption.getPresentedOfferingContext()
        );
    }

}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/SubscriptionOptionAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.PresentedOfferingContext
import com.revenuecat.purchases.models.GoogleInstallmentsInfo
import com.revenuecat.purchases.models.GoogleSubscriptionOption
import com.revenuecat.purchases.models.InstallmentsInfo
import com.revenuecat.purchases.models.PricingPhase
import com.revenuecat.purchases.models.SubscriptionOption

@Suppress("unused", "UNUSED_VARIABLE", "deprecation")
private class SubscriptionOptionAPI {
    fun checkSubscriptionOption(subscriptionOption: SubscriptionOption) {
        val phases: List<PricingPhase> = subscriptionOption.pricingPhases
        val tags: List<String> = subscriptionOption.tags
        val isBasePlan: Boolean = subscriptionOption.isBasePlan
        val presentedOfferingId: String? = subscriptionOption.presentedOfferingIdentifier
        val presentedOfferingContext: PresentedOfferingContext? = subscriptionOption.presentedOfferingContext
        val isPrepaid: Boolean = subscriptionOption.isPrepaid
        val installmentsInfo: InstallmentsInfo? = subscriptionOption.installmentsInfo
    }

    fun checkGoogleSubscriptionOption(googleSubscriptionOption: GoogleSubscriptionOption) {
        checkSubscriptionOption(googleSubscriptionOption)
        val productId = googleSubscriptionOption.productId
        val basePlanId = googleSubscriptionOption.basePlanId
        val offerId = googleSubscriptionOption.offerId
        val offerToken = googleSubscriptionOption.offerToken
        val productDetails = googleSubscriptionOption.productDetails
        val installmentsInfo: GoogleInstallmentsInfo? = googleSubscriptionOption.installmentsInfo

        val subscriptionOption = GoogleSubscriptionOption(
            productId,
            basePlanId,
            offerId,
            googleSubscriptionOption.pricingPhases,
            googleSubscriptionOption.tags,
            productDetails,
            offerToken,
        )

        val subscriptionOptionWithOfferingId = GoogleSubscriptionOption(
            productId,
            basePlanId,
            offerId,
            googleSubscriptionOption.pricingPhases,
            googleSubscriptionOption.tags,
            productDetails,
            offerToken,
            "offeringId",
        )

        val subscriptionOptionWithOfferingContext = GoogleSubscriptionOption(
            productId,
            basePlanId,
            offerId,
            googleSubscriptionOption.pricingPhases,
            googleSubscriptionOption.tags,
            productDetails,
            offerToken,
            googleSubscriptionOption.presentedOfferingContext,
        )

        val subscriptionOptionWithNullOfferingContext = GoogleSubscriptionOption(
            productId,
            basePlanId,
            offerId,
            googleSubscriptionOption.pricingPhases,
            googleSubscriptionOption.tags,
            productDetails,
            offerToken,
            null,
        )

        val subscriptionOptionWithInstallmentsInfo = GoogleSubscriptionOption(
            productId,
            basePlanId,
            offerId,
            googleSubscriptionOption.pricingPhases,
            googleSubscriptionOption.tags,
            productDetails,
            offerToken,
            presentedOfferingContext = null,
            installmentsInfo,
        )
    }
}
